app/src/main/java/swen_anigans/mathematicfanatic/QuizActivity.java

```diff
@@ -42,22 +42,70 @@ protected void onCreate(Bundle savedInstanceState) {
             initializeQuizQuestions();
 
             for (int i = 0; i < totalPages; i++) {
-                answers.set(i, 0);
-                expectedAnswers.set(i, -1);
+                answers.add(0);
+                expectedAnswers.add(-1);
             }
         }
 
+        /*

```

Comment:
This is the timing code that was causing crashes. I commented it for now so that I could come back to it tomorrow morning. The commented code below is to prevent duplicate questions. It wasn't working as expected, and we agreed that it wasn't high priority